### ISSUE TITLE
Bug(`sort-imports`): Side effect grouping errors not reported in specific cases

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#316]
### Describe the bug

The code below is considered valid when it should not:

### Code example

Configuration:
```
{
  "sortSideEffects": false,
  "newlinesBetween": "always",
  "groups": [
    "side-effect-style",
    "side-effect"
  ]
}
```

Code:
```ts
import '~/data'
import '~/css/globals.css'
```

Autofixing does work however: triggering it through other means, such as 

```ts
import '~/data'
import '~/css/globals.css'

import b from "b"
import a from "a"
```

will correctly sort the code to the expected
```ts
import '~/css/globals.css'

import '~/data'

import a from "a"
import b from "b"
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.8.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.
